### PR TITLE
change cloning workflow (WP-647)

### PR DESCRIPTION
**Pull request by vsolovei-smartling**
When cloning submissions, they previously applied XML immediatelyat upload job run. This could cause issues when a posts relation was not yet processed, resulting in copying the relation ids instead of altering them as needed. After this PR, every cloned submission first creates target content, then XML is applied with all existing relations having proper target ids.

### CODE REVIEW

File: inc/Smartling/Jobs/UploadJob.php [hunk 1]
```diff
@@ -58,6 +60,18 @@ private function processUploadQueue(): void
...
+    private function processCloning(): void
+    {
+        $submissions = $this->submissionManager->findSubmissionsForCloning();
+        foreach ($submissions as $submission) {
+            do_action(ExportedAPI::ACTION_SMARTLING_PREPARE_SUBMISSION_UPLOAD, $submission);
+        }
+        // needs to be in a separate loop to have all relations when cloning
+        foreach ($submissions as $submission) {
+            do_action(ExportedAPI::ACTION_SMARTLING_CLONE_CONTENT, $submission);
+        }
```
Comment (PavelLoparev) [inline inc/Smartling/Jobs/UploadJob.php:72]: Do we really need to first process ALL submissions with `ACTION_SMARTLING_PREPARE_SUBMISSION_UPLOAD`? I wonder if we can have just one loop that calls `ACTION_SMARTLING_PREPARE_SUBMISSION_UPLOAD` -> `ACTION_SMARTLING_CLONE_CONTENT` in one iteration

**Review by PavelLoparev — COMMENTED** [review]
Left some comments. I got the initial idea but I'm not sure will it break something or not (I'm surprised not so many tests are changed after such code change)

File: inc/Smartling/Jobs/UploadJob.php (hunk 1 above)
Comment (vsolovei-smartling) [reply inc/Smartling/Jobs/UploadJob.php:72]: Yes, that's the idea for this pull requests. First create all target content, then change the ids as necessary.

Comment (PavelLoparev) [reply inc/Smartling/Jobs/UploadJob.php:72]: I mean, why not:
```
foreach ($submissions as $submission) {
            do_action(ExportedAPI::ACTION_SMARTLING_PREPARE_SUBMISSION_UPLOAD, $submission);
             do_action(ExportedAPI::ACTION_SMARTLING_CLONE_CONTENT, $submission);
        }
        ```

**Pull request merged**